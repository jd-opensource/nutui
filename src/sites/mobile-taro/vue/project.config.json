{
  "miniprogramRoot": "./dist",
  "projectname": "@nutui/nutui-taro-mobile",
  "description": "nutui-taro-vue",
  "appid": "touristappid",
  "setting": {
    "urlCheck": true,
    "es6": false,
    "postcss": false,
    "minified": false
  },
  "compileType": "miniprogram"
}
