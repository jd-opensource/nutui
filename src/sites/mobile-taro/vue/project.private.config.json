{
  "setting": {},
  "condition": {
    "plugin": {
      "list": []
    },
    "game": {
      "list": []
    },
    "gamePlugin": {
      "list": []
    },
    "miniprogram": {
      "list": [
        {
          "name": "business/pages/address/index",
          "pathName": "business/pages/address/index",
          "query": "",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/noticebar/index",
          "pathName": "feedback/pages/noticebar/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/progress/index",
          "pathName": "feedback/pages/progress/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/swiper/index",
          "pathName": "feedback/pages/swiper/index",
          "query": "",
          "scene": null
        }
      ]
    }
  }
}