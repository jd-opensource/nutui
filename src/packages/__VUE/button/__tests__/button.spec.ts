import { mount } from '@vue/test-utils';
import Button from '../index.vue';
import { nextTick } from 'vue';
import { StarFill } from '@nutui/icons-vue';
test('emit click event', () => {
  const wrapper = mount(Button);

  wrapper.trigger('click');
  expect(wrapper.emitted('click')!.length).toEqual(1);
});

test('slot test', async () => {
  const wrapper = mount(Button, {
    slots: {
      default: '按钮测试'
    }
  });
  expect(wrapper.html()).toContain('按钮测试');
});
test('should emit click event', () => {
  const wrapper = mount(Button);

  wrapper.trigger('click');
  expect(wrapper.emitted('click')).toHaveLength(1);
});

test('should not allow click when set disabled props', async () => {
  const wrapper = mount(Button, {
    props: {
      disabled: true
    }
  });
  wrapper.trigger('click');
  await nextTick();
  expect(wrapper.emitted('click')).toBeFalsy();
});
test('should not emit click event when loading', () => {
  const wrapper = mount(Button, {
    props: {
      loading: true
    }
  });

  wrapper.trigger('click');
  expect(wrapper.emitted('click')).toBeFalsy();
});
test('should find nut-icon class of an svg tag when using icon slot', () => {
  const wrapper = mount(Button, {
    slots: {
      icon: StarFill
    }
  });

  const icons = wrapper.findAll('svg');
  expect(icons.length).toEqual(1);
  expect(icons[0].element.classList).toContain('nut-icon');
});
