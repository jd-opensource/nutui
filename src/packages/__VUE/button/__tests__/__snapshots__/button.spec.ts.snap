// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change icon class prefix when using icon-class-prefix prop 1`] = `
"<view class=\\"nut-button nut-button--default nut-button--normal nut-button--round\\">
  <view class=\\"nut-button__warp\\">
    <!--v-if-->
    <!---->
    <!--v-if-->
  </view>
</view>"
`;
